socless/events.py

```diff
@@ -94,215 +142,211 @@ def dedup_hash(self):
+        )
+        playbook_input_as_dict = asdict(playbook_input)
 
-    def execute_playbook(self, entry, investigation_id=""):
-        """Execute a playbook for a SOCless event.
-        Args:
-            entry (dict): The event details
-            investigation_id (str): The investigation_id to use
-            playbook (str): The name of the playbook to execute
-        Returns:
-            dict: The execution_id, investigation_id and a status indicating if the playbook
-                execution request was successful
-        """
-        meta = {"investigation_id": investigation_id, "playbook": self.playbook}
-        if not investigation_id:
-            investigation_id = gen_id()
-        RESULTS_TABLE = os.environ.get("SOCLESS_RESULTS_TABLE")
-        playbook_input = {"artifacts": {}, "results": {}, "errors": {}}
-        playbook_arn = "arn:aws:states:{region}:{accountid}:stateMachine:{stateMachineName}".format(
-            region=os.environ["AWS_REGION"],
-            accountid=self.lambda_context.invoked_function_arn.split(":")[4],
-            stateMachineName=self.playbook,
+        setup_results_table_for_playbook(
+            self.metadata.execution_id,
+            self.metadata.investigation_id,
+            playbook_input_as_dict,
         )
-        execution_id = gen_id()
-        playbook_input["artifacts"]["event"] = entry
-        playbook_input["artifacts"]["execution_id"] = execution_id
-        results_table = boto3.resource("dynamodb").Table(RESULTS_TABLE)
-        _ = results_table.put_item(
-            Item={
-                "execution_id": execution_id,
-                "datetime": gen_datetimenow(),
-                "investigation_id": investigation_id,
-                "results": playbook_input,
-            }
+
+        report = StartExecutionReport(
+            investigation_id=self.metadata.investigation_id,
+            playbook=str(self.event.playbook),
+            statemachinearn=playbook_arn,
+            execution_id=self.metadata.execution_id,
+            error="",
         )
-        stepfunctions = boto3.client("stepfunctions")
         try:
-            _ = stepfunctions.start_execution(
-                name=execution_id,
+            _ = stepfunctions_client.start_execution(
+                name=self.metadata.execution_id,
                 stateMachineArn=playbook_arn,
-                input=json.dumps(
-                    {
-                        "execution_id": execution_id,
-                        "artifacts": playbook_input["artifacts"],
-                    }
-                ),
-            )
-            socless_log.info(
-                "Playbook execution started",
-                dict(
-                    meta,
-                    **{"statemachinearn": playbook_arn, "execution_id": execution_id},
-                ),
+                input=json.dumps(playbook_input_as_dict),
             )
+            socless_log.info("Playbook execution started", report.__dict__)
```

Comment:
#question: is calling the `__dict__` method here the only way to get out the data we need?

Reply:
converting a dataclass to a dict can be done two different ways (besides manually building the dict:
- __dict__ is fast but isn't recursive. if one of your dataclass attributes is another class/dataclass, its not gonna work correctly
- dataclasses.asdict(object) is ~ 10x slower but works recursively.